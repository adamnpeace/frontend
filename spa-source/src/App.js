import React, { Component } from "react";
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import logo from "./img/logo.svg";

import { Col, Row } from 'reactstrap';
import auth0Client from './Auth';
import NavBar from './NavBar/NavBar';
import ClassSelector from "./ClassSelector/ClassSelector";

class App extends Component {
  render() {
    return (
      <div>
        <NavBar />
        <br />
        <Row>
          <Col xs={0} md={1} />
          <Col xs={12} md={10}>
            <Route exact path="/" exact component={Home} />
            <Route path="/editor" component={ClassSelector} />
          </Col>
          <Col xs={0} md={1} />
        </Row>
      </div>
    );
  }
}

function Home() {
  return (
    <div>
      <h2>Home</h2>
      Welcome to the mlAcademy test site | This is not a production build
    </div>
  );
}

function About({ match }) {
  return (
    <div>
      <h2>About</h2>
      Wow, another learning platform
    </div>
  );
}

export default App;