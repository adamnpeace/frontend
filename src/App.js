import React from "react";
import LabView from "./LabView/LabView";
import { Box, Heading, Grommet, ResponsiveContext } from "grommet";
import Header from "./Header";

const theme1 = {
  global: {
    colors: {
      brand: "#7D4CDB",
      "accent-1": "#6FFFB0",
      "accent-3": "#90fcf9"
    },
    font: {
      family: "Roboto",
      size: "14px",
      height: "20px"
    }
  }
};

const theme2 = {
  global: {
    colors: {
      brand: "#000000",
      "accent-1": "#6FFFB0",
      "accent-3": "#90fcf9"
    },
    font: {
      family: "Roboto",
      size: "14px",
      height: "20px"
    }
  }
};

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      theme: theme1
    }
  }

  changeTheme() {
    var newTheme = this.state.theme === theme1 ? theme2 : theme1;
    this.setState({theme: newTheme});
  }

  render() {
    return (
      <Grommet theme={this.state.theme} full>
        <ResponsiveContext.Consumer>
          {size => (
            <div>
              <Header>
                <Heading level="3" margin="none">
                  mlAcademy
                </Heading>
              </Header>
              <Box
                height="100%"
                background="bg"
                elevation="none"
                fill="vertical"
                margin="small"
              >
                <LabView classID={8} />
              </Box>
            </div>
          )}
        </ResponsiveContext.Consumer>
      </Grommet>
    );
  }
}

export default App;
