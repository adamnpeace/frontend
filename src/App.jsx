import React from 'react';
import { Box, Button, Grommet, ResponsiveContext } from 'grommet';
import { Link, Route, withRouter } from 'react-router-dom';
import axios from 'axios';
import PropTypes from 'prop-types';

import Header from './components/Header';
import logoTextWhite from './img/logo_text_white.svg';
import Home from './views/Home';
import Lab from './views/Lab';
import Login from './views/Login';
import Topics from './views/Topics';
import { Config, theme1, theme2 } from './Config';

const propTypes = {
  history: PropTypes.object.isRequired
};
const defaultProps = {};

const LabLoader = ({ height, match: { params } }) => {
  const { id } = params;
  return <Lab height={height} topicID={id} />;
};

LabLoader.propTypes = { height: PropTypes.number.isRequired, match: PropTypes.object.isRequired };

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      isAuthenticated: false,
      theme: theme1,
      topicID: 0,
      lessonID: 0,
      classes: [],
      height: 0
    };
    this.updateWindowDimensions = this.updateWindowDimensions.bind(this);
  }

  componentDidMount() {
    this.updateWindowDimensions();
    window.addEventListener('resize', this.updateWindowDimensions);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateWindowDimensions);
  }

  updateWindowDimensions() {
    this.setState({ height: window.innerHeight });
  }

  changeTheme() {
    this.setState(prevState => ({ theme: prevState.theme === theme1 ? theme2 : theme1 }));
  }

  goTo(route) {
    this.props.history.push(route);
  }

  handlePrev() {
    this.setState(prevState => ({ lessonID: prevState.lessonID - 1 }));
  }

  handleNext() {
    this.setState(prevState => ({ lessonID: prevState.lessonID + 1 }));
  }

  render() {
    const { height } = this.state;
    return (
      <Grommet theme={this.state.theme} full>
        <Header>
          <Link to="/">
            <img src={logoTextWhite} alt="Logo" height="40pt" />
          </Link>

          <Box direction="row" gap="small">
            <Button
              color="accent-4"
              label="home"
              onClick={() => {
                this.goTo('/');
              }}
              primary
            />
            <Button
              color="accent-4"
              label="learn"
              onClick={() => {
                this.goTo('/topics');
              }}
              primary
            />
            {!this.state.isAuthenticated && (
              <Button
                color="accent-4"
                label="login"
                onClick={() => {
                  this.goTo('/login');
                }}
              />
            )}
          </Box>
        </Header>
        <div style={{ height: `${height - 60}px` }}>
          <Route exact path="/" component={Home} />
          <Route path="/topics" render={() => <Topics />} />
          <Route
            path="/labs/:id"
            render={props => <LabLoader height={this.state.height} {...props} />}
          />
          <Route path="/login" component={Login} />
        </div>
      </Grommet>
    );
  }
}

App.propTypes = propTypes;
App.defaultProps = defaultProps;

export default withRouter(App);
