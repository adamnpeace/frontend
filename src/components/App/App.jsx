import React from 'react';
import Helmet from 'react-helmet';
import Router from '../routes/Router';
import Site from './Site';
import Header from '../Header/Header';
import Content from './Content';
import { info } from 'getConstants';

function App(props) {
  const { name, tagline } = info;
  return (
    <Site>
      <Helmet
        title={name}
        meta={[
          {
            name: 'description',
            content: { tagline }
          },
          {
            name: 'keywords',
            content:
              'machine, learning, data, science, computers, computing, computer, science, high, school, game, tutorial'
          }
        ]}
      />
      <Header />
      <Content>
        <Router />
      </Content>
    </Site>
  );
}

App.propTypes = {};

export default App;
