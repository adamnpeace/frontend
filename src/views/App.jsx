import React from 'react';
import { Box, Button, Grommet, Layer } from 'grommet';
import { Link, Route, Switch, withRouter } from 'react-router-dom';
import Notifications, { notify } from 'react-notify-toast';
import PropTypes from 'prop-types';
import { Header, NoMatch, TooSmall } from './components';
import logoTextWhite from '../img/logo_text_white.svg';
import Home from './Home';
import Lab from './Lab';
import Login from './auth/Login';
import Register from './auth/Register';
import Topics from './Topics';
import { theme } from '../Config';
import Firebase from './auth/firebase';

const propTypes = {
  history: PropTypes.object.isRequired
};
const defaultProps = {};

const LabLoader = ({ height, match: { params } }) => {
  const { id } = params;
  return <Lab height={height} topicID={id} />;
};

LabLoader.propTypes = { height: PropTypes.number.isRequired, match: PropTypes.object.isRequired };

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isAuthenticated: false,
      height: 0,
      width: 0
    };
    this.updateWindowDimensions = this.updateWindowDimensions.bind(this);
  }

  componentDidMount() {
    this.updateWindowDimensions();
    window.addEventListener('resize', this.updateWindowDimensions);
    Firebase.isInitialized().then(val => {
      this.setState({ isAuthenticated: val });
    });
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateWindowDimensions);
  }

  updateWindowDimensions() {
    this.setState({ height: window.innerHeight, width: window.innerWidth });
  }

  handleLogout() {
    Firebase.logout();
    this.setState({ isAuthenticated: false });
    notify.show('You have been logged out successfully!', 'warning');
  }

  render() {
    const { height, width, isAuthenticated } = this.state;
    const { history } = this.props;
    const name = Firebase.getCurrentUsername();
    return (
      <Grommet theme={theme} full>
        <Notifications />
        <Header>
          <Link to="/">
            <img src={logoTextWhite} alt="Logo" height="40pt" />
          </Link>

          <Box direction="row" gap="small">
            <Button
              color="accent-4"
              label="home"
              onClick={() => {
                history.replace('/');
              }}
              primary
            />

            {isAuthenticated ? (
              <Button
                color="accent-4"
                label="learn"
                onClick={() => {
                  history.replace('/topics');
                }}
                primary
              />
            ) : (
              <Button
                color="accent-4"
                label="learn"
                onClick={() => {
                  history.replace('/login');
                }}
                primary
              />
            )}
            {isAuthenticated ? (
              <Button
                color="accent-3"
                primary
                label={`Hi ${name}`}
                onClick={() => {
                  this.handleLogout();
                  this.props.history.replace('/');
                }}
              />
            ) : (
              <Button
                color="accent-4"
                label="login"
                onClick={() => {
                  history.replace('/login');
                }}
              />
            )}
          </Box>
        </Header>
        <div style={{ height: `${height - 60}px` }}>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/topics" render={() => <Topics />} />
            <Route
              path="/labs/:id"
              render={props => <LabLoader height={this.state.height} {...props} />}
            />
            <Route path="/login" component={Login} />
            <Route path="/register" component={Register} />
            <Route component={NoMatch} />
          </Switch>
        </div>
        {width < 840 && <TooSmall />}
      </Grommet>
    );
  }
}

App.propTypes = propTypes;
App.defaultProps = defaultProps;

export default withRouter(App);
