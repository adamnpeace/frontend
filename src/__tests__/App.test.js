import React from "react";
import ReactDOM from "react-dom";
import App from '../App';
import { StaticRouter } from 'react-router';
import renderer from "react-test-renderer";

describe("Entire App Test", () => {
  const context = {};

  it("Entire app renders without crashing", () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <StaticRouter location="location" context={context}>
        <App />
      </StaticRouter>, div);
  });

  it("Entire app renders correctly according to first snapshot", () => {
    const component = renderer.create(
      <StaticRouter location="location" context={context}>
        <App />
      </StaticRouter>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
