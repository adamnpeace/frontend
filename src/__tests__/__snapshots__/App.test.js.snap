// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entire App Test Entire app renders correctly according to first snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 bcjtJw"
>
  <div>
    <header
      className="StyledBox-sc-13pk1d4-0 bRQYIV"
      style={
        Object {
          "zIndex": "1",
        }
      }
    >
      <img
        alt="Logo"
        height="40pt"
        onClick={[Function]}
        src="logo_text_white.svg"
      />
      <div
        className="StyledBox-sc-13pk1d4-0 ffeAql"
      >
        <button
          className="StyledButton-sc-323bzc-0 gQJkQe"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          type="button"
        >
          Home
        </button>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iChEkS"
        />
        <button
          className="StyledButton-sc-323bzc-0 gQJkQe"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          type="button"
        >
          Labs
        </button>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 iChEkS"
        />
        <button
          className="StyledButton-sc-323bzc-0 iigCHW"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          type="button"
        >
          Login
        </button>
      </div>
    </header>
  </div>
</div>
`;
